### PR TITLE
Use dev container for integration tests

### PR DESCRIPTION
**Pull request by GernotMaier**
addresses #600 .

Use `simtools-dev:latest` for integration tests.

This avoids the compilation of corsika/sim_telarray and the installation of all python packages for each tests.

Reduced run time for tests from 25-28 min to 15-18 min.

Notes:

- only one python version is tested (the one installed the container). This should be fine, as unit tests should pick up any differences between the versions
- future pull requests which updates packages require to manually call the docker-dev deployment workflow (as these packages are no in simtools-dev:latest). This should be fine, as we very seldomly add new packages.

### CODE REVIEW

**Comment by GernotMaier** [comment]
There is some success. 

It seems now that there is no database access. Not sure if we have to open a port or something else.

**Review by orelgueta — APPROVED** [review]

**Comment by orelgueta** [comment]
> * future pull requests which updates packages require to manually call the docker-dev deployment workflow (as these packages are no in simtools-dev:latest). This should be fine, as we very seldomly add new packages.

What do you mean by "new packages" here?

**Comment by GernotMaier** [comment]
> > * future pull requests which updates packages require to manually call the docker-dev deployment workflow (as these packages are no in simtools-dev:latest). This should be fine, as we very seldomly add new packages.
> 
> What do you mean by "new packages" here?

I mean if we add a dependency to environment.yml in a PR, this new dependency will not be available for the integration tests.

**Comment by orelgueta** [comment]
Ahh I see, thanks!

File: .github/workflows/CI-integrationtests.yml [hunk 1]
```diff
@@ -19,76 +19,28 @@ on:
...
-            python-version: "3.9"
-            install-method: mamba
-
-          - os: ubuntu-latest
-            python-version: "3.9"
-            install-method: pip
+    runs-on: ubuntu-latest
+    container:
+      image: ghcr.io/gammasim/simtools-dev:latest
+      options: --user 0
```
[the remark this answers is not in the document]
Comment (GernotMaier) [reply .github/workflows/CI-integrationtests.yml:24]: Honestly I am not sure - I have been trying without and it didn't work. This was an outcome of a search on the internet... (I know, not ideal...)

[the remark this answers is not in the document]
Comment (orelgueta) [reply .github/workflows/CI-integrationtests.yml:24]: OK, that's reasonable enough.

**Pull request merged**